prow/create-build-cluster.sh

```diff
@@ -0,0 +1,154 @@
+  prompt "Generate kubeconfig credentials for Prow" gencreds
+  echo "All done!"
+}
+# Prep and check args.
+function parseArgs() {
+  for var in TEAM PROJECT ZONE CLUSTER MACHINE NODECOUNT DISKSIZE FOLDER_ID BILLING_ACCOUNT_ID; do
+    if [[ -z "${!var}" ]]; then
+      echo "Must specify ${var} environment variable (or specify a default in the script)."
+      exit 2
+    fi
+    echo "${var}=${!var}"
+  done
+}
+function prompt() {
+  local msg="$1" cmd="$2"
+  echo
+  read -r -n1 -p "$msg ? [y/n] "
+  echo
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    "$cmd"
+  else
+    echo "Skipping and continuing to next step..."
+  fi
+}
+function pause() {
+  read -n 1 -s -r
+}
+
+authed=""
+function getClusterCreds() {
+  if [[ -z "${authed}" ]]; then
+    gcloud container clusters get-credentials --project="${PROJECT}" --zone="${ZONE}" "${CLUSTER}"
+    authed="true"
+  fi
+}
+function createProject() {
+  # Create project, configure billing, enable GKE, add IAM rule for oncall team.
+  echo "Creating project '${PROJECT}' (this may take a few minutes)..."
+  gcloud projects create "${PROJECT}" --name="${PROJECT}" --folder="${FOLDER_ID}"
+  gcloud beta billing projects link "${PROJECT}" --billing-account="${BILLING_ACCOUNT_ID}"
+  gcloud services enable "container.googleapis.com" --project="${PROJECT}"
+  gcloud projects add-iam-policy-binding "${PROJECT}" --member="${ADMIN_IAM_MEMBER}" --role="roles/owner"
+}
+function createCluster() {
+  echo "Creating cluster '${CLUSTER}' (this may take a few minutes)..."
+  echo "If this fails due to insufficient project quota, request more at https://console.cloud.google.com/iam-admin/quotas?project=${PROJECT}"
+  echo
+  gcloud container clusters create "${CLUSTER}" --project="${PROJECT}" --zone="${ZONE}" --machine-type="${MACHINE}" --num-nodes="${NODECOUNT}" --disk-size="${DISKSIZE}" --disk-type="pd-ssd" --enable-autoupgrade --enable-autorepair
+  getClusterCreds
+  kubectl create namespace "test-pods"
+}
+function createUploadSASecret() {
+  getClusterCreds
+  local sa="prow-pod-utils"
+  local saFull="${sa}@${PROJECT}.iam.gserviceaccount.com"
+  # Create a service account for uploading to GCS.
+  gcloud beta iam service-accounts create "${sa}" --project="${PROJECT}" --description="SA for Prow's pod utilities to use to upload job results to GCS." --display-name="Prow Pod Utilities"
+  # Generate private key and attach to the service account.
+  gcloud iam service-accounts keys create "sa-key.json" --project="${PROJECT}" --iam-account="${saFull}"
+  kubectl create secret generic "service-account" -n "test-pods" --from-file="service-account.json=sa-key.json"
+  echo
+  echo "Please ask the test-infra oncall (https://go.k8s.io/oncall) to run the following:"
+  echo "  gsutil acl ch -u \"${saFull}:O\" \"gs://${GCS_BUCKET}\""
+  echo
+  echo "Press any key to aknowledge (this doesn't need to be completed to continue this script, but it needs to be done before uploading will work)..."
```

Comment:
Consider splitting into two shorter log lines:
```
echo "This must be done before uploading will work"
echo "However, the script can continue (press any key)..."
```
Consider splitting the occurrence in `gencreds()` also.

nit: missing 'c' in acknowledge